Survival analysis of HGG tumor subtypes

jharenza opened the pull request:
❗ staggered on #1264 but I cannot change the base to that branch

<!--Hi there, thanks for your contribution! Please take a moment to fill out this template to facilitate the review of your pull request.-->

### Purpose/implementation Section

#### What scientific question is your analysis addressing?
This PR addresses bullet 3 in [this comment](https://github.com/AlexsLemonade/OpenPBTA-analysis/pull/1264#pullrequestreview-942730959), in which we would like to perform a univariate analysis of subtype within the HGG tumors. This was previously done in v17, but for only H3 WT and DMG K28, so was quite outdated.

#### What was your approach?
I renamed the old notebook which did HGG subtype analysis and performed both logrank and cox regression (univariate) for molecular subtype.


#### What GitHub issue does your pull request address?
NA: bullet 3 in [this comment](https://github.com/AlexsLemonade/OpenPBTA-analysis/pull/1264#pullrequestreview-942730959)


### Directions for reviewers. Tell potential reviewers what kind of feedback you are soliciting.

#### Which areas should receive a particularly close look?
double check everything


#### Is there anything that you want to discuss further?
I am not sure why the subtypes which are `DMG, H3 K28` are not showing up in the results.  I ran a double check of the formula manually, and get the same results. There are more samples in this group than in some other groups, so I am baffled. Any ideas @sjspielman ?

#### Is the analysis in a mature enough form that the resulting figure(s) and/or table(s) are ready for review?
yes


### Results

#### What types of results are included (e.g., table, figure)?
- KM plot and logrank results table for hgg subtypes
- cox results table for hgg subtypes

#### What is your summary of the results?
Here I go screen-shotting again...
<img width="1171" alt="Screen Shot 2022-04-20 at 2 24 01 PM" src="https://user-images.githubusercontent.com/207034/164297502-83316046-2590-423d-871c-3b1a6443a5f4.png">
HGG tumor subtypes with K28, TP53 loss have a HR of 1.15 (0.41-1.88), p = 0.002, supporting current knowledge that co-occurrence of TP53 with K28 results in worse prognosis. 

#### Reproducibility Checklist

<!-- Check all those that apply or remove this section if it is not applicable.-->

- [x] The dependencies required to run the code in this pull request have been added to the project Dockerfile.
- [x] This analysis has been added to continuous integration.

#### Documentation Checklist

<!-- Please review https://github.com/AlexsLemonade/OpenPBTA-analysis#documenting-your-analysis -->

- [ ] This analysis module has a `README` and it is up to date.
- [ ] This analysis is recorded in the table in `analyses/README.md` and the entry is up to date.
- [x] The analytical code is documented and contains comments.

<!-- IF YOUR PULL REQUEST IS A DATA RELEASE, PLEASE REMOVE THE [HTML COMMENT TAG](https://html.com/tags/comment-tag/) FROM THE SECTION BELOW AND COMPLETE THE CHECKLIST-->

<!-- 
### Data Release Checklist
- [ ] Is the table in doc/data-file-descriptions.md up to date?
- [ ] Is doc/data-format.md up to date?
- [ ] Is doc/release-notes.md up to date?
- [ ] Is download-data.sh up to date?
- [ ] Was download-data.sh tested and did it complete without error?
-->

sjspielman commented:
@jharenza thanks for working on this!!

Had a look about the missing level. It's "there"! For the cox regression, it's the _reference category_. So, every coefficient can be interpreted as relative to the reference.

```
# Whatever is first is, by default, the reference! Or, you can change the factor levels
levels(factor(metadata$molecular_subtype))
[1] "DMG, H3 K28"                 "DMG, H3 K28, TP53 activated" "DMG, H3 K28, TP53 loss"     
[4] "HGG, H3 wildtype"            "HGG, H3 wildtype, TP53 loss"
```

To get pairwise comparisons among these, you can use [pairwise_survdiff](https://rdrr.io/cran/survminer/man/pairwise_survdiff.html). I went ahead and added a chunk in there so you can pull and see.

jharenza commented:
> @jharenza thanks for working on this!!
> 
> Had a look about the missing level. It's "there"! For the cox regression, it's the _reference category_. So, every coefficient can be interpreted as relative to the reference.
> 
> ```
> # Whatever is first is, by default, the reference! Or, you can change the factor levels
> levels(factor(metadata$molecular_subtype))
> [1] "DMG, H3 K28"                 "DMG, H3 K28, TP53 activated" "DMG, H3 K28, TP53 loss"     
> [4] "HGG, H3 wildtype"            "HGG, H3 wildtype, TP53 loss"
> ```
> 
> To get pairwise comparisons among these, you can use [pairwise_survdiff](https://rdrr.io/cran/survminer/man/pairwise_survdiff.html). I went ahead and added a chunk in there so you can pull and see.

Oh, this is perfect. Thank you!

jharenza commented:
@sjspielman this is ready now, and this would be the panel H forest plot I was envisioning in number 2 [here](https://github.com/AlexsLemonade/OpenPBTA-analysis/issues/1323)

jharenza commented:
Just another two notes.
1. I added some code to check for rna-only samples, since they were subtyped. But, they were all non- Initial CNS tumor, so they were not added.
2. I didn't keep it in, but I did check making the reference `HGG, H3 wildtype`, and while almost everything is significant in the hazard plot (all K28 and TP53 subtypes), a weird thing was happening in which there was no reference plotted and `HGG, H3 wildtype`, even though it was the ref, it did not align with HR 1, and showed significance. No idea why, so I just removed this because the more useful comparisons are using H3 K28 as reference.

sjspielman commented on analyses/survival-analysis/survival-analysis_subtypes.Rmd:
```diff
@@ -99,40 +110,44 @@ This way we subset the metadata to 1 single primary sample per patient ID
+           !is.na(OS_years)) %>%
+  unique()
+
+rna_pts_to_add <- setdiff(unique(metadata_rna$Kids_First_Participant_ID), metadata_ind_dna$Kids_First_Participant_ID)
```
This line is a confusing - There is a `#nope` comment, but when I print this variable out (and it should be printed in the notebook if we need to see the results to get to that "nope" conclusion), it has output consisting of 5 patient ids:

```
> rna_pts_to_add <- setdiff(unique(metadata_rna$Kids_First_Participant_ID), metadata_ind_dna$Kids_First_Participant_ID)
> rna_pts_to_add
[1] "PT_63HDZHCN" "PT_9BZETM0M" "PT_0SPKM4S8" "PT_0MXPTTM3" "PT_S0Q27J13"
```

Based on this seems to me there _are_ RNA patients not in the DNA subset?

jharenza replied:
I thought these were not primary tumors or maybe not RNA-only samples, but I can check again and should update the code for this.

jharenza replied:
hmm, I still get `character(0)` - printing and will commit

sjspielman replied:
Hm, let me see again..

sjspielman commented on analyses/survival-analysis/survival-analysis_subtypes.Rmd:
```diff
@@ -99,40 +110,44 @@ This way we subset the metadata to 1 single primary sample per patient ID
 # find molecular subtypes which have more than n_participants (3) participants
-filter_molecular_subtype <- metadata %>% group_by(molecular_subtype) %>% dplyr::tally() %>% dplyr::filter(n>n_participants)
+filter_molecular_subtype <- metadata_ind_dna %>% group_by(molecular_subtype) %>% dplyr::tally() %>% dplyr::filter(n>n_participants)
 
-# select only molecular_subtypes that have more than n_participants (3) participants
-metadata <- metadata %>%
+# select only  that have more than n_participants (3) participants
+metadata_ind_dna <- metadata_ind_dna %>%
   dplyr::filter(molecular_subtype %in%
                   filter_molecular_subtype$molecular_subtype)
```
This can be simplified to only 1 line of code:

```
metadata_ind_dna <- metadata_ind_dna %>%
  group_by(molecular_subtype) %>% 
  # directly add a column for the counts instead of tallying down into a summarized tibble
  mutate(n_molecular_subtype = n()) %>%
  # keep only sufficient n
  filter(n_molecular_subtype>n_participants) %>%
  # rm the counting column and it's all cleaned up
  select(-n_molecular_subtype)
```

sjspielman replied:
We still have two groups going into modeling that only have 1 sample each: `HGG, IDH, TP53 activated ` and `HGG, H3 G35`. That means there is no variation in the groups to model. Hence, we see them coming up with _both_ CI's as `NA` and an unreliable estimate.

On one hand I would not want to include groupings that have insufficient data, but on the other hand if the goal is to compare among groups we cannot cherry-pick them. My personal assessment here is that there is not sufficient data to rely on the model output.

jharenza replied:
I was wondering why we didn't see the G35 in the final table and it is because there aren't enough samples per the n=3 per subtype rule set in place in this notebook prior to my edits. I think some are also not primary tumors, so I'll check why they're even in the model..?

jharenza replied:
I am not seeing these in the model either

sjspielman commented on analyses/survival-analysis/survival-analysis_subtypes.Rmd:
````diff
@@ -167,11 +182,43 @@ Save the plot to a file.
+               file.path(results_dir, "cox_hgg_subtype.RDS"))
+```
+
+Why are we not seeing H3 K28 subtypes?
````
Since this subtype is in the model, this should be rephrased or chunk updated accordingly.

sjspielman commented on analyses/survival-analysis/survival-analysis_subtypes.Rmd:
````diff
@@ -167,11 +182,43 @@ Save the plot to a file.
+Why are we not seeing H3 K28 subtypes?
+```{r cox manual check}
+# Double check the function isn't missing them - get the same result below
+metadata_recode <- metadata_ind_dna %>%
````
this line in the chunk is needed for `pairwise_survdiff`, so I'd recommend just 1 chunk here with recoding, survdiff, and the plot.

jharenza commented:
@jaclyn-taroni thank you for helping with `ungroup()`!! @sjspielman this is ready for re-review

sjspielman commented on analyses/survival-analysis/survival-analysis_subtypes.Rmd:
```diff
@@ -167,12 +194,29 @@ Save the plot to a file.
-cowplot::save_plot(filename = kap_meier_plot_file, plot = surv_plot_all_subtype)
+cowplot::save_plot(filename = kap_meier_plot_file, plot = surv_plot_all_subtype, base_height = 5, base_width = 8)
 # Let's save the model for CI and median survial as well
 readr::write_tsv(surv_median(kap_fit$model), kap_meier_model_file)
```
We should update this to and RDS (which will include changing the `kap_meier_model_file` variable name!) so that _all_ model information is saved. 
```suggestion
readr::write_rds(kap_fit$model, kap_meier_model_file)
```

jharenza replied:
yes, good idea

sjspielman reviewed (commented):
Looks good! Noting to update based on [this comment](https://github.com/AlexsLemonade/OpenPBTA-analysis/pull/1348#issuecomment-1110085619) - set up a separate directory for the result files.

jharenza commented:
> Looks good! Noting to update based on [this comment](https://github.com/AlexsLemonade/OpenPBTA-analysis/pull/1348#issuecomment-1110085619) - set up a separate directory for the result files.

done!

sjspielman reviewed (commented):
Looks like some files went a bit awry at some point here. The files exported by the template Rmd are either deleted in this PR or got deleted elsewhere.

I'm going to add some commits *here* to restore those files. As part of this, I'm updating the template Rmd to export result files to `results/template/` for consistency with what other notebooks are now going to do.

sjspielman commented:
I had approved this, but realized one additional change needs to be made - we also need to save the data that went into the KM model to make the figure for the paper. So, instead of exporting `kap_fit$model` to RDS, it now exports `kat_fit` to RDS.

This previously passed CI right before this update, and this additional line of code I added here was confirmed to run in Docker locally with a fresh environment. Therefore this is being merged before checks are complete.
